src/lib/components/InputGroup/README.mdx

```diff
@@ -0,0 +1,228 @@
+          label="Variety"
+          placeholder="Eg. Golden delicious"
+        />
+        <Button label="Submit" />
+      </InputGroup>
+    );
+  }}
+</Playground>
+
+See [API](#api) for all available options.
+
+## General Guidelines
+
+Use input group to group **related fields and actions** that a user can
+take. Input fields and buttons should not be grouped just to save space on the
+screen.
+
+### Shared Property
+
+You can set size property directly on InputGroup to be shared for all fields
+and buttons inside the group. This property is then passed over to individual
+elements. At the same time, it **cannot be overridden** on the fields' or
+buttons' level. While technically possible, from the design point of view it's
+undesirable to group elements of totally different types or sizes.
+
+## Sizes
+
+All existing field and button sizes are also available on the input group level:
+small, medium, and large.
+
+<Playground>
+  <InputGroup
+    label="Small size"
+    size="small"
+  >
+    <TextField label="Input" />
+    <Button label="Submit" />
+  </InputGroup>
+  <InputGroup label="Medium size">
+    <TextField label="Input" />
+    <Button label="Submit" />
+  </InputGroup>
+  <InputGroup
+    label="Large size"
+    size="large"
+  >
+    <TextField label="Input" />
+    <Button label="Submit" />
+  </InputGroup>
+</Playground>
+
+## Invisible Label
+
+In some cases, it may be convenient to visually hide the group label. The label
+remains accessible to assistive technologies.
+
+While it may be acceptable for login screens with just a few fields or for other
+simple forms, it's dangerous to hide labels from users in most cases. Keep in
+mind you should **provide another visual clue** so users know what to fill into
+the input.
+
+<Playground>
+  <InputGroup
+    isLabelVisible={false}
+    label="First and last name"
+  >
+    <TextField
+      label="First name"
+      placeholder="Eg. John"
+    />
+    <TextField
+      label="Last name"
+      placeholder="Eg. Doe"
+    />
+    <Button label="Submit" />
+  </InputGroup>
+</Playground>
+
+## Vertical layout
+
+The default horizontal layout is very easy to use and work with. However, there
+are situations where vertical layout suits better — and that's why React UI
+supports this kind of layout as well.
+
+<Playground>
+  <InputGroup
+    label="Vertical layout"
+    layout="vertical"
+  >
+    <TextField label="Label" />
+    <Button label="Submit" />
+  </InputGroup>
+</Playground>
+
+## States
+
+### Validation States
+
+Validation states visually present the result of validation of the grouped
+inputs. Input group's validation state is taken from its child inputs. You
+should always **provide validation messages for states other than valid**
+directly through `validationTexts` prop so users know what happened and what
+action they should take or what options they have. Validation messages passed
+to input elements' `validationText` prop will be ignored.
+
+<Playground>
```

Comment:
I would like to provide en example with multiple `validationTexts`.

Also, as specified in the issue, the validation texts are not semantically tied to the input elements, the connection should be expressed in textual form in the actual message.

Could you please update the example to better illustarte this?